src/modules/ApprovalVotingModule.sol

```diff
@@ -124,45 +127,40 @@ contract ApprovalVotingModule is VotingModule {
                 if (option.targets.length != option.values.length || option.targets.length != option.calldatas.length) {
                     revert InvalidParams();
                 }
-            }
 
-            // Push proposal options in storage
-            for (uint256 i; i < optionsLength; ++i) {
-                _proposals[proposalId].options.push(proposalOptions[i]);
+                proposals[proposalId].options.push(option);
             }
         }
 
-        _proposals[proposalId].governor = msg.sender;
-        _proposals[proposalId].settings = proposalSettings;
-        _proposals[proposalId].optionVotes = new uint128[](optionsLength);
+        proposals[proposalId].governor = msg.sender;
+        proposals[proposalId].settings = proposalSettings;
+        proposals[proposalId].optionVotes = new uint128[](optionsLength);
     }
 
     /**
-     * Count approvals voted by `account`. If voting for, options need to be set in ascending order.
+     * Count approvals voted by `account`. If voting for, options need to be set in ascending order. Votes can only be cast once.
      *
      * @param proposalId The id of the proposal.
      * @param account The account to count votes for.
-     * @param support The type of vote to count. 0 = For, 1 = Abstain.
+     * @param support The type of vote to count.
      * @param weight The weight of the vote.
      * @param params The ids of the options to vote for sorted in ascending order, encoded as `uint256[]`.
      */
     function _countVote(uint256 proposalId, address account, uint8 support, uint256 weight, bytes memory params)
         external
         override
     {
-        Proposal memory proposal = _proposals[proposalId];
+        Proposal memory proposal = proposals[proposalId];
         _onlyGovernor(proposal.governor);
-
-        if (hasVoted(proposalId, account)) revert VoteAlreadyCast();
-
-        uint128 weight_ = weight.toUint128();
+        if (accountVotes[proposalId][account] != 0) revert AlreadyVoted();
```

Comment:
Is this just extra safety check? Because in the `OptimismGovernorV5.sol#_castVote()` function, we do the normal `_countVote()`, which checks `hasVoted`.

Reply:
There's some context here:

- The module is technically designed to be used with any governor, not just the related OP governor
- The option of allowing re-casting votes was being recently discussed, both for governor and module

This check prevents the module from being used with re-voting, even when the underlying governor might support it. Performing the check before `if (support == uint8(VoteType.For))` also ensures this is prevented regardless of the `VoteType` cast.